examples/typeclass-tests.dx

```diff
@@ -0,0 +1,38 @@
+interface InterfaceTest1 a:Type where
+  InterfaceTest1 : a
+> Error: variable already defined: InterfaceTest1
+
+interface InterfaceTest2 typeName:Type where
+  typeName : typeName -> typeName
+
+interface InterfaceTest3 _:Type where
+  foo : Int32
+
+> Parse error:8:26:
+>   |
+> 8 | interface InterfaceTest3 _:Type where
+>   |                          ^^^^^
+> unexpected "_:Typ"
```

Comment:
I have to say that this message is pretty confusing. I don't even understand why does this case cause a failure

Reply:
It is not strictly necessary to throw an error here, but the proposal to disallow anonymous type variables comes from https://github.com/google-research/dex-lang/pull/271#discussion_r509364060, hence why this throws an error. Let me know what you think about that, it is of course still possible to revert (and then reenable `mkVarTypeName`)

Reply:
ooh, i bet it's because "where" is 5 characters long, so the parser is checking the first 5 characters of `_:Type where` to see if it matches "where", and then those first 5 characters somehow make it into the error message

Reply:
yeah the error message is very weird, and I'd like us to fix that. About type parameters without names, I updated the thread you linked to. I think it's perfectly reasonable to allow that, but I guess we can leave that for later too.